src/services/utils.hooks.js

```diff
@@ -0,0 +1,60 @@
+const checkPermissions = require('feathers-permissions');
+const { preventChanges } = require('feathers-hooks-common');
+
+
+/**
+ * Checks wheter requested user is admin or owns resource.
+ * WARNING: So far it works only for `users` endpoint.
+ */
+function checkAdminOrOwner() {
+    return checkPermissions({
+        roles(context) {
+            const { user } = context.params;
```

Comment:
You're repeating this code three times in three functions:

```
const { user } = context.params;
     if (user !== undefined) {
          if (user.username == context.arguments[0]) {
```

Please refactor it.